### PR TITLE
Fix: generate docs for Terraform ComponentDefinition

### PR DESCRIPTION
**Pull request by zzxwill**
Generated kubevela.io docs for Terraform typed ComponentDefinition
with `vela def gen-doc xxx` cli.

Signed-off-by: Zheng Xi Zhou <zzxwill@gmail.com>

### Description of your changes

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: references/cli/def.go [hunk 1]
```diff
@@ -325,6 +327,40 @@ func NewDefinitionGetCommand(c common.Args) *cobra.Command {
 	return cmd
 }
 
+// NewDefinitionGenDocCommand create the `vela def gen-doc` command to generate documentation of definitions
+func NewDefinitionGenDocCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "gen-doc NAME",
+		Short: "Generate documentation of definitions (Only Terraform typed definitions are supported)",
```
Comment (wonderflow) [inline references/cli/def.go:334]: can we also support other cue based definition?

Comment (zzxwill) [reply references/cli/def.go:334]: Yes, it's in my plan. This feature is the official feature by CLI to generate docs for KubeVela definitions (this was supported by https://github.com/oam-dev/kubevela/blob/master/hack/references/generate.go). I tried to generate all definitions (including workflowstep), but 1) there were some blockers to run the tool, 2) the generated docs differs from thoes hosted in kubevela.io. So I plan to support Terraform typed definition first, and then for all components, traits, policy and workflow. Is that okay?

File: pkg/controller/utils/capability.go [hunk 2]
```diff
@@ -190,9 +190,16 @@ func GetOpenAPISchemaFromTerraformComponentDefinition(configuration string) ([]b
 // GetTerraformConfigurationFromRemote gets Terraform Configuration(HCL)
 func GetTerraformConfigurationFromRemote(name, remoteURL, remotePath string) (string, error) {
 	tmpPath := filepath.Join("./tmp/terraform", name)
+	// Check if the directory exists. If yes, remove it.
```
Comment (zzxwill) [inline pkg/controller/utils/capability.go:193]: @FogDong Here are two modifications of this method. Please help check whether they are okay with you.

File: references/cli/def.go (hunk 1 above)
Comment (wonderflow) [reply references/cli/def.go:334]: sure,thanks

**Review by wonderflow — APPROVED** [review]

**Review by Somefive — APPROVED** [review]

**Pull request merged**